lib/private/files/storage/wrapper/encryption.php

```
@@ -441,6 +451,130 @@ public function fopen($path, $mode) {
 		return $this->storage->fopen($path, $mode);
 	}
 
+
+	/**
+	 * make some plausibility checks if the the unencrypted size is correct.
+	 * If not, we calculate the correct unencrypted size and return it
+	 *
+	 * @param string $path internal path relative to the storage root
+	 * @param int $size size of the physical file
+	 * @param int $unencryptedSize size of the unencrypted file
+	 * @param int $headerSize size of the header
+	 * @param array $header key-value pair with the header content
+	 * @param IEncryptionModule $encryptionModule
+	 *
+	 * @return int unencrypted size
+	 */
+	protected function verifyUnencryptedSize(
+		$path,
+		$size,
+		$unencryptedSize,
+		$headerSize,
+		$header,
+		IEncryptionModule $encryptionModule
+	) {
+		if ($unencryptedSize < 0 ||
+			($size > 0 && $unencryptedSize === $size)
+		) {
+			return $this->fixUnencryptedSize($path, $size, $unencryptedSize, $headerSize, $header, $encryptionModule);
+		}
+
+		return $unencryptedSize;
+	}
+
+	/**
+	 * calculate the unencrypted size
+	 *
+	 * @param string $path internal path relative to the storage root
+	 * @param int $size size of the physical file
+	 * @param int $unencryptedSize size of the unencrypted file
+	 * @param int $headerSize size of the header
+	 * @param array $header key-value pair with the header content
+	 * @param IEncryptionModule $encryptionModule
+	 *
+	 * @return int calculated unencrypted size
+	 */
+	protected function fixUnencryptedSize(
+		$path,
+		$size,
+		$unencryptedSize,
+		$headerSize,
+		$header,
+		IEncryptionModule $encryptionModule
+	) {
+
+		$stream = $this->storage->fopen($path, 'r');
+
+		// if we couldn't open the file we return the old unencrypted size
+		if (!is_resource($stream)) {
+			return $unencryptedSize;
+		}
```

Not sure if returning a wrong result is the best thing to do here. Why not return a meaningful error?